src/common/type/stringId.h

```diff
@@ -0,0 +1,123 @@
+/***********************************************************************************************************************************
+Represent Short Strings as Integers
+
+It is often useful to represent identifiers as strings when they cannot easily be represented as an enum/integer, e.g. because they
+are distributed among a number of unrelated modules or need to be passed to remote processes. Strings are also more helpful in
+debugging since they can be recognized without cross-referencing the source. However, strings are awkward to work with in C since
+they cannot be directly used in switch statements leading to less efficient if-else structures.
+
+A StringId encodes a short string into an integer so it can be used in switch statements but may also be readily converted back into
+a string for debugging purposes. StringIds may also be suitable for matching user input providing the strings are short enough.
+
+See strIdGenerate() for information on StringId constants.
+***********************************************************************************************************************************/
+#ifndef COMMON_TYPE_STRINGID_H
+#define COMMON_TYPE_STRINGID_H
+
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "common/type/string.h"
+
+/***********************************************************************************************************************************
+Maximum number of characters in the string representation of a StringId, including the + prefix indicator when present. This is a
+safe buffer size when calling strIdToZN. If the buffer needs to be zero-terminated then an extra byte should be added.
+***********************************************************************************************************************************/
+// Maximum for specific encodings (e.g. 5-bit, 6-bit)
+#define STRID5_MAX                                                  13
+#define STRID6_MAX                                                  11
+
+// Maximum for any encoding
+#define STRID_MAX                                                   STRID5_MAX
+
+/***********************************************************************************************************************************
+StringId typedef to make them more recognizable in the code
+***********************************************************************************************************************************/
+typedef uint64_t StringId;
+
+/***********************************************************************************************************************************
+Number of bits to use for encoding. The number of bits affects the character set that can be encoded.
+***********************************************************************************************************************************/
+typedef enum
+{
+    stringIdBit5 = 0,                                               // 5-bit encoding for a-z, 1-4, and - characters
```

Comment:
Will this be used for options? What about reset-pg8-host-cmd - how will it be able to be used with an 8 and also how will we be able to distinguish reset-pg8-host when 13 characters are max (or if it must use STRID6, that would be even worse, no?)

Reply:
There is no plan to use StringIds for option names.